homeassistant/util/package.py

```diff
@@ -30,22 +30,30 @@ def is_docker_env() -> bool:
     return Path("/.dockerenv").exists()
 
 
-def is_installed(package: str) -> bool:
+def get_installed_versions(specifiers: set[str]) -> set[str]:
+    """Return a set of installed packages and versions."""
+    return {specifier for specifier in specifiers if is_installed(specifier)}
+
+
+def is_installed(requirement_str: str) -> bool:
     """Check if a package is installed and will be loaded when we import it.
 
+    expected input is a pip compatible package specifier
+    e.g. "package==1.0.0" or "package>=1.0.0,<2.0.0"
+
     Returns True when the requirement is met.
     Returns False when the package is not installed or doesn't meet req.
     """
     try:
-        distribution(package)
-        return True
```

Comment:
This hasn't worked for a long time since distributions don't accept `==` and it would always raise `PackageNotFound` now